### ISSUE TITLE
Create Heroku app and infrastructure for MIT Open

### ISSUE THREAD

**Issue by blarghmatey** [mitodl/ol-infrastructure#1657]
# User Story

- As a developer working on MIT Open I want to be able to deploy my changes to a shared environment for testing and validation
- As an MIT Open product owner/stakeholder I want to be able to review and test features in the application

# Description/Context
We are deploying a new copy of MIT Open/Open Discussions to facilitate development of the new product vision. To that end we need a Heroku application and associated AWS infrastructure set up that mimics the current open-discussions environment.

# Acceptance Criteria

- [x] There is a Heroku application pipeline configured to deploy from https://github.com/mitodl/mit-open
- [ ] There is a Salt proxy configured to manage settings for the Heroku pipeline
- [ ] There is a Pulumi project for managing the AWS infrastructure needed by MIT Open
- [x] This also includes the OpenSearch cluster so that we can keep the two runtimes completely separate.

**Comment by shaidar** [comment]
Was gonna start populating vault values, however the mount hasn't been created. Looked at this PR https://github.com/mitodl/ol-infrastructure/commit/4b8f1510c3c5230207688ce16dd20d53688a9965 but just wasn't sure where to start and didn't want to override any previous work. Will work on getting Touchstone working with Keycloak.

**Comment by shaidar** [comment]
We need to decide on a domain name to use for the Heroku managed TLS cert and we also need to setup a new mailgun domain

**Comment by Ardiea** [comment]
@rhysyngsun We added you as a collaborator to the `mitopen-rc` app in Heroku. I believe that salt-ops code _wants very badly_ to update the config vars for that app but it cannot because the build is failing with:

```


-----> Generating cache tables
       Traceback (most recent call last):
         File "/app/manage.py", line 12, in <module>
           from django.core.management import execute_from_command_line
       ModuleNotFoundError: No module named 'django'

```

I reconciled the buildpacks vs. `odl-open-dicussions-rc` and those look good but I noticed that the detected "frameworks" for `mitopen-rc` is just Node.js while there are a bunch listed for `odl-open-discussions-rc`. I'm not sure where those come from? `app.json`. 

Let me know if you want to get together at some point and get it further along.
